wandb/sdk/wandb_settings.py

```diff
@@ -738,11 +738,16 @@ def _validate_anonymous(value: str) -> bool:
             raise UsageError(f"Settings field `anonymous`: '{value}' not in {choices}")
         return True
 
-    @staticmethod
-    def _validate_api_key(value: str) -> bool:
+    def _validate_api_key(self, value: str) -> bool:
         if len(value) > len(value.strip()):
             raise UsageError("API key cannot start or end with whitespace")
+
+        if value.startswith("local") and self.base_url == "https://api.wandb.ai":
```

Comment:
don't we have a test for local... I would use it? in case we want to change the way we test local

Reply:
Do you mean we should use `self.is_local` instead? Or were you referring to the `test_local*` tests in `test_cli.py`?
Anyways, I think we should be using `is_local` here, changed.

Reply:
yeah meant is_local